# แบบฝึกหัด 6

---

In [ ]:
import numpy as np
import pandas as pd
from scipy import stats

6.1 จากการสำรวจโดยหน่วยงานของรัฐ ความน่าจะเป็นที่คนไทยจะมีบัตรเครดิตอย่างน้อย 1 บัตรเท่ากับ 0.6 หากสุ่มตัวอย่างคนไทยจำนวน
1000 คน จงประมาณความน่าจะเป็นที่คนไทยอย่างน้อย 615 คน จากที่ถูกเลือกเป็นตัวอย่าง จะมีบัตร
เครดิตอย่างน้อย 1 บัตร ด้วยการแจกแจงแบบปกติ

In [ ]:
#แสดงวิธีทำ

*X* แทนจำนวนคนไทยที่ถูกเลือกเป็นตัวอย่างจะมีบัตร เครดิตอย่างน้อย 1 บัตร   
*n* จำนวนคนไทยที่สุ่มมาทดสอบ *n*=1000   
p  ความน่าจะเป็นที่คนไทยจะมีบัตรเครดิตอย่างน้อย 1 บัตร  p=0.6  
  
𝑋~𝑏(𝑥;𝑛=1000, p=0.6)

In [ ]:
n=1000
p=0.6
q=1-p

In [ ]:
#หาค่า n*p
n*p

600.0

In [ ]:
#หาค่า n*q
n*q

400.0

𝑛p และ 𝑛q มากกว่า 5 ใช้การประมาณด้วยการแจกแจงแบบปกติได้  
𝑌~𝑁(𝜇=𝑛p=600, 𝜎^2=𝑛pq=240)

ประมาณความน่าจะเป็นที่คนไทยอย่างน้อย 615 คน จากที่ถูกเลือกเป็นตัวอย่าง จะมีบัตร เครดิตอย่างน้อย 1 บัตร  
คือหาค่า P(X≥615)  
ที่ X≥615 คือ Y>614.5

In [ ]:
#แปลง 614.5 เป็น Z และหาค่าความน่าจะเป็นที่คนไทยอย่างน้อย 615 คน จากที่ถูกเลือกเป็นตัวอย่าง จะมีบัตร เครดิตอย่างน้อย 1 บัตร
stats.norm.sf((614.5-600)/np.sqrt(240))

0.17464406593420118

ความน่าจะเป็นที่คนไทยอย่างน้อย 615 คน จากที่ถูกเลือกเป็นตัวอย่าง จะมีบัตร เครดิตอย่างน้อย 1 บัตร คือ 0.175

###   


6.2 บริษัทแห่งหนึ่งสุ่มเครื่อง server มา 22 เครื่อง พบว่ามีค่าบำรุงรักษาเฉลี่ยเครื่องละ 10000 บาทต่อปี และมีส่วนเบี่ยงเบนมาตราฐานของค่าบำรุงรักษา 5000 บาทต่อปี 
หากการวางแผนงบประมาณจะอ้างอิงจากค่าบำรุงรักษาโดยเฉลี่ยของ server จากทั้งบริษัท จงหาความน่าจะเป็นที่ค่าบำรุงรักษา server โดยเฉลี่ยทั้งบริษัทจะมีค่าน้อยกว่า 12000 บาท  
หากกำหนดให้การแจกแจงของค่าบำรุงรักษาจากกลุ่มตัวอย่างเป็นแบบที

In [ ]:
#แสดงวิธีทำ

𝑛=22,  ∴ 𝑣=22−1=21  จำนวนตัวอย่าง 𝑛<30 ไม่ทราบ 𝜇 และ 𝜎 ใช้การแจกแจงแบบที  
x̄=10000,  𝑠=5000  
ต้องการหา  P(𝜇<12000)

In [ ]:
#คำนวณคะแนนที
t_cal = (10000-12000) / (5000 / np.sqrt(22))
t_cal

-1.876166303929372

โจทย์ข้อนี้หาความน่าจะเป็นเมื่อ 𝜇<12000 ซึ่งหมายความว่าเมื่อ t> -1.876

In [ ]:
#ความน่าจะเป็นที่ค่าบำรุงรักษา server โดยเฉลี่ยทั้งบริษัทจะมีค่าน้อยกว่า 12000 บาท
stats.t.sf(-1.876, df=21)

0.9626874014857781

ความน่าจะเป็นที่ค่าบำรุงรักษา server โดยเฉลี่ยทั้งบริษัทจะมีค่าน้อยกว่า 12000 บาทคือ 0.963

###   


6.3 จากข้อ 6.2 จงหาช่วงของค่าบำรุงรักษาโดยเฉลี่ยของ server จากทั้งบริษัทที่ทำให้  
P(-a < T < a) = 0.99  โดย a คำนวณจากค่าสถิติของตัวอย่าง server ในข้อ 6.2

In [ ]:
#แสดงวิธีทำ

เนื่องจากการแจกแจงของค่าบำรุงรักษาจากกลุ่มตัวอย่างเป็นแบบที  
P(-a < T < a) = 0.99 

ดังนั้น  
α = (1 - 0.99)/2

In [ ]:
#หาค่า a
a = stats.t.isf((1 - 0.99)/2, df=21)
a

2.831359558017186

In [ ]:
a

2.831359558017186

In [ ]:
stats.t.ppf((1 - 0.99)/2, df=21)

-2.831359558017186

เมื่อทราบค่า a จะสามารถหาค่า 𝜇 ได้ จากสูตร a = ( x̄-𝜇 )/ (s/√𝑛)

In [ ]:
mu1 = 10000 - ((-a)*5000/np.sqrt(22))
mu1

13018.239430147843

In [ ]:
mu2 = 10000 - (a*5000/np.sqrt(22))
mu2

6981.760569852157

ช่วงของค่าบำรุงรักษาโดยเฉลี่ยของ server จากทั้งบริษัทที่ทำให้
P(-a < T < a) = 0.99 คือ 13018.239 > 𝜇 > 6981.761